src/EnergyPlus/GroundHeatExchangers.cc

```diff
@@ -1217,498 +2230,825 @@ namespace GroundHeatExchangers {
 			}
 		}
 
 	}
 
 	//******************************************************************************
 
-	void
+	Real64
+	GLHEVert::calcBHAverageResistance()
+	{
+		// Calculates the average thermal resistance of the borehole using the first-order multipole method.
+
+		// Javed, S. & Spitler, J.D. 2016. 'Accuracy of Borehole Thermal Resistance Calculation Methods
+		// for Grouted Single U-tube Ground Heat Exchangers.' Applied Energy.187:790-806.
+
+		// Equation 13
+
+		Real64 const beta = 2 * Pi * grout.k * calcPipeResistance();
+
+		Real64 const final_term_1 = log( theta_2 / ( 2 * theta_1 * pow( 1 - pow_4( theta_1 ), sigma ) ) );
+		Real64 const num_final_term_2 = pow_2( theta_3 ) * pow_2( 1 - ( 4 * sigma * pow_4( theta_1 ) ) / ( 1 - pow_4( theta_1 ) ) );
+		Real64 const den_final_term_2_pt_1 = ( 1 + beta ) / ( 1 - beta );
+		Real64 const den_final_term_2_pt_2 = pow_2( theta_3 ) * ( 1 + ( 16 * sigma * pow_4( theta_1 ) ) / pow_2( 1 - pow_4( theta_1 ) ) );
+		Real64 const den_final_term_2 = den_final_term_2_pt_1 + den_final_term_2_pt_2;
+		Real64 const final_term_2 = num_final_term_2 / den_final_term_2;
+
+		return ( 1 / ( 4 * Pi * grout.k ) ) * ( beta + final_term_1 - final_term_2 );
+
+	}
+
+	//******************************************************************************
+
+	Real64
+	GLHEVert::calcBHTotalInternalResistance()
+	{
+		// Calculates the total internal thermal resistance of the borehole using the first-order multipole method.
+
+		// Javed, S. & Spitler, J.D. 2016. 'Accuracy of Borehole Thermal Resistance Calculation Methods
+		// for Grouted Single U-tube Ground Heat Exchangers.' Applied Energy. 187:790-806.
+
+		// Equation 26
+
+		Real64 beta = 2 * Pi * grout.k * calcPipeResistance();
+
+		Real64 final_term_1 = log( pow( 1 + pow_2( theta_1 ), sigma ) / ( theta_3 * pow( 1 - pow_2( theta_1 ), sigma ) ) );
+		Real64 num_term_2 = pow_2( theta_3 ) * pow_2( 1 - pow_4( theta_1 ) + 4 * sigma * pow_2( theta_1 ) );
+		Real64 den_term_2_pt_1 = ( 1 + beta ) / ( 1 - beta ) * pow_2( 1 - pow_4( theta_1 ) );
+		Real64 den_term_2_pt_2 = pow_2( theta_3 ) * pow_2( 1 - pow_4( theta_1 ) );
+		Real64 den_term_2_pt_3 = 8 * sigma * pow_2( theta_1 ) * pow_2( theta_3 ) * ( 1 + pow_4( theta_1 ) );
+		Real64 den_term_2 = den_term_2_pt_1 - den_term_2_pt_2 + den_term_2_pt_3;
+		Real64 final_term_2 = num_term_2 / den_term_2;
+
+		return ( 1 / ( Pi * grout.k ) ) * ( beta + final_term_1 - final_term_2 );
+	}
+
+	//******************************************************************************
+
+	Real64
+	GLHEVert::calcBHGroutResistance()
+	{
+		// Calculates grout resistance. Use for validation.
+
+		// Javed, S. & Spitler, J.D. 2016. 'Accuracy of Borehole Thermal Resistance Calculation Methods
+		// for Grouted Single U-tube Ground Heat Exchangers.' Applied Energy. 187:790-806.
+
+		// Equation 3
+
+		return calcBHAverageResistance() - calcPipeResistance() / 2.0;
```

Comment:
I'm sure you're fine here, but just the look of this equation screams that it's wrong 😆